Stanimir

StanimirIliev opened the pull request:
@mlesikov Should I open a pull request for you to see the changes or it is just for me to see the changes when doing merge?

StanimirIliev committed:
Deleted unnecessery text files

StanimirIliev committed:
Changed packages of all files

mlesikov commented:
hm, actaally with PR is easier changes to be made before they go to the master with single commit.
For these first tasks, maybe it will be harder with PR, because they are small and you will have to swich between too many baranches, but it is better to practice this workflow - it is standart workflow for development

mlesikov commented:
hm, actaally with PR is easier changes to be made before they go to the master with single commit.
For these first tasks, maybe it will be harder with PR, because they are small and you will have to swich between too many baranches, but it is better to practice this workflow - it is standart workflow for development

mlesikov commented:
It is essential the PR to contains changes only for one task - specific changes, that later can be combined in a single commit

mlesikov commented on .idea/misc.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
project files of the IDE (inteliIDEA) should not be part of the repository, because they contains specific settings for your warking station, they can be add .gitignore file. Later when you get to the build tools chapter, you will find some new tools that are used

StanimirIliev replied:
Ok I will remove all files of the IDE. Should I do merge PR or just to open it and wait you to see it?

mlesikov replied:
I should approve the PR 
![image](https://user-images.githubusercontent.com/210499/28318306-f2d02cca-6bd2-11e7-94dd-aa2b5a4ce04e.png)
then if no conflicts, at the bottom of the PR front page you will see

![image](https://user-images.githubusercontent.com/210499/28318341-1cf176e4-6bd3-11e7-98b4-7cfa3c661256.png)

but this will be enable and then you can merge it with the master.
I use the word "merge" , but we newer really make a merges, we use rebase and merge.
rebase is a spcial command that applies the commits of your bracnch over the commits of the master and do not make a "merge commit" like this
![image](https://user-images.githubusercontent.com/210499/28318428-7d4eb5c4-6bd3-11e7-89f1-1a31c5c34f59.png)
so the commits in the master looks clear and order. 
You should check what debase is and ho to do it, also you should check and how to resolve conflicts. This tools help many team members to review and merge their daily work.

mlesikov reviewed (changes requested):
test request changes - now when somone requests changes the PR could not be merged

mlesikov commented:
as you can see am a reviewer and I am requesting changes 
![image](https://user-images.githubusercontent.com/210499/28318676-4bd284d4-6bd4-11e7-982a-f6466b774ddc.png)

StanimirIlev commented:
So when I make changes I should make PR, but when I make the new chapters and add files to the repositoriy should I push them by the master branch and avoid PR?

StanimirIliev committed:
Add git ignore file

mlesikov commented:
The workflow is that every change should pass code review - this means that PR should be added that contains that change and should be reviewed by some of the team mates. That;s why these PRs are created. There is no difference if the changes contains fixes or new functionalities.
Like that Open source projects are being developed by a whole community of interested developers.

For example, if I would like to add changes to your master, but I am not a contributor, I can fork your repository and then I can create a branch from your master, then I can add my changes and create a PR in your repository with the branch I worked into. Then you can review only the cahnges and then you can accept them to your master.(That is how open source projects are being developed)

Also, as you can see, we can easily add comments on this exact PR and the discussion could be for the exact changes.

mlesikov commented:
At this time it will be a little harder for you to add every change with the PR, but will be easier for me to review them. After a while, this will become easier to you and I hope so you will like it.

StanimirIlev commented:
Ok I will do PR's

StanimirIliev committed:
Changed gitignore file

StanimirIliev committed:
Deleted files of the IDE

mlesikov commented:
I will approve this PR, I suggest you to merge it with the master and to start a new branch (for_exemple object_in_java_task_1 - working branch) and to add changes for that task. Something else that will be helpfull - when you create PR, the github gets the first commit message and suggest it as a title and a body of the PR - so if you cover format that is described in the post that I share with you, you will have no problem to the title and the description of the PR

StanimirIliev committed:
Chapter: Object in Java, Task: Sumator

Changed method: public static int sum(String a, String b)
Now when the parameters are not valid integers, the program throws
NumberFormatException, instead of to read from the console

mlesikov commented:
https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history
this is from the chapter Repositories
